base/uk.ac.stfc.isis.ibex.ui.blocks/src/uk/ac/stfc/isis/ibex/ui/blocks/groups/BlocksMenu.java

```diff
@@ -160,6 +141,10 @@ public void menuAboutToShow(IMenuManager manager) {
 
         appendToGroup(BLOCK_MENU_GROUP, logSubMenu);
         
+        createEditBlockLabelAndAction();
+        
+        appendToGroup(BLOCK_MENU_GROUP, editBlockAction);
```

Comment:
Just noticed, I don't think we don't actually need `editBlockAction` as an instance variable anymore?
```suggestion
        final var editBlockAction= createEditBlockLabelAndAction();
        
        appendToGroup(BLOCK_MENU_GROUP, editBlockAction);
```

And make `createEditBlockLabelAndAction()` return the action (as `IAction`) and remove `this.editBlockAction`?

Reply:
like this? https://github.com/ISISComputingGroup/ibex_gui/pull/1456/commits/746ba8ae7e5ae88550be3cd20a4e6aeefc2172cf

Reply:
That code change is what I meant, yeah. I assume you've retested it just to check I haven't suggested something stupid...

Reply:
yh, it still works